### PR TITLE
CONCF-769 | CuratedContent/GameGuides setting/purging cache

### PR DESCRIPTION
**Pull request by Warkot**
https://wikia-inc.atlassian.net/browse/CONCF-769

Opening a new request that ports changes from #7666 (and carries it on) after changes to dev on #7713.

### CODE REVIEW

File: extensions/wikia/MercuryApi/MercuryApiHooks.class.php [hunk 1]
```diff
@@ -4,6 +4,7 @@ class MercuryApiHooks {
 
 	const SERVICE_API_BASE = '/api/v1/';
 	const SERVICE_API_ARTICLE = 'article/';
+	const SERVICE_API_CURATED_CONTENT = 'curatedContent/';
```
Comment (rogatty) [inline extensions/wikia/MercuryApi/MercuryApiHooks.class.php]: It's `/main/section/{sectionName}` and `/main/category/{categoryName}` (https://github.com/Wikia/mercury/blob/dev/server/routes.ts#L105-L114).

We should purge the category URL too, right?

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 2]
```diff
@@ -65,14 +45,11 @@ public function getPage() {
 		global $wgTitle;
 
 		// This will always return json
-		$this->response->setFormat( 'json' );
-
-		$this->cacheResponseFor( 7, self::DAYS );
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+		$this->response->setCacheValidity( WikiaResponse::CACHE_STANDARD );
```
Comment (vforge) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: Yay for standarization!

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 3]
```diff
@@ -502,13 +492,65 @@ private function getCuratedContentQualityForWiki( $wikiID ) {
 	 *
 	 * @return bool
 	 */
-	static function onCuratedContentSave( $sections ) {
-		self::purgeMethod( 'getList' );
```
Comment (vforge) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: So.. As you're removing this call - is `self::purgeMethod` is being used anywhere?

Comment (Warkot) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: `self::purgeMethod` is inherited from `self`(`CuratedContentController`) → `WikiaController` → `WikiaDispatchableObject`. It's used in few places. I am purging this URL in one `SquidUpdate` → line 528.
```
// Purge all sections list getter URL - no additional params
[ self::getUrl( 'getList' ) ]
```

File: extensions/wikia/MercuryApi/MercuryApiHooks.class.php (hunk 1 above)
Comment (Warkot) [inline extensions/wikia/MercuryApi/MercuryApiHooks.class.php]: Yup. The URLs I can think of now we have to purge are:
* Ember calls to Hapi:
  * the ones you mentioned above,
* Hapi calls to MediaWiki:
  * `/wikia.php?controller=MercuryApi&method=getArticle&title=${Main_Page}` https://github.com/Wikia/mercury/blob/dev/server/lib/MediaWiki.ts#L92, 
  * all of `/wikia.php?controller=MercuryApi&method=getCuratedContentSection&section=${section}` https://github.com/Wikia/mercury/blob/dev/server/lib/MediaWiki.ts#L118,
  * all of `/wikia.php?controller=ArticlesApi&method=getList&…&category=${categoryName}` https://github.com/Wikia/mercury/blob/dev/server/lib/MediaWiki.ts#L128,
* first static pages generated by Hapi e.g.:
  * http://elderscrolls.wikia.com/wiki/The_Elder_Scrolls_Wiki (do we need http://elderscrolls.wikia.com/ too?),
  * http://elderscrolls.wikia.com/main/section/TES%2520Online,
  * http://elderscrolls.wikia.com/main/category/Skyrim.
* native apps calls:
  * `/wikia.php?controller=ArticlesApi&method=getList` which should remain cached and purged,
  * all of `/wikia.php?controller=ArticlesApi&method=getList&section=${section}` which caching and purging is already added in this PR.
  * I'm wondering how apps get category entries and how it's cached/purged if at all.

…so much has to be added and/or rewritten. :wrench:

**Comment by rogatty** [comment]
Let's remember about fixing https://github.com/Wikia/app/pull/7802/files#diff-2dc5a31ce84c2146a550d826db0f31f2R504

**Comment by vforge** [comment]
LGTM

**Comment by bognix** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
LGTM

**Pull request merged**